src/__experimental__/components/date/date-picker.component.js

```diff
@@ -44,10 +45,11 @@ const DatePicker = (props) => {
   };
 
   useEffect(() => {
-    if (props.selectedDate && monthOrYearHasChanged(datepicker, props.selectedDate)) {
-      datepicker.current.showMonth(props.selectedDate);
+    if (hasComponentUpdated(currentInputDate, isoFormattedValueString(props.inputDate))) {
```

Comment:
there is no need to use that method arguments, as they could be accessed in it's body
```suggestion
    if (hasComponentUpdated()) {
```

Reply:
Good point will fix locally